modules/emoteevBidAdapter.js

```diff
@@ -153,62 +422,70 @@ export const getDeviceInfo = (deviceDimensions, viewDimensions, documentDimensio
+  else if (config.debug) return config.debug;
+  else return false;
+};
 
+/**
+ * EmoteevBidAdapter spec
+ * @access public
+ * @type {BidderSpec}
+ */
 export const spec = {
   code: BIDDER_CODE,
   supportedMediaTypes: [BANNER],
-
-  isBidRequestValid: (bid) => {
-    return !!(
-      bid &&
-      bid.params &&
-      bid.params.adSpaceId &&
-      bid.bidder === BIDDER_CODE &&
-      validateSizes(bid.mediaTypes.banner.sizes)
-    );
-  },
-
-  buildRequests: (validBidRequests, bidderRequest) => {
-    const payload = Object.assign({},
-      {
-        akPbjsVersion: AK_PBJS_VERSION,
-        bidRequests: validBidRequests.map(conformBidRequest),
-        currency: config.getConfig('currency'),
-        debug: emoteevDebug(utils.getParameterByName('emoteevDebug'), config.getConfig('emoteev.debug')),
-        language: navigator.language,
-        refererInfo: bidderRequest.refererInfo,
-        deviceInfo: getDeviceInfo(getDeviceDimensions(), getViewDimensions(), getDocumentDimensions(), isWebGLEnabled()),
-        userAgent: navigator.userAgent,
-      },
-      emoteevOverrides(utils.getParameterByName('emoteevOverrides'), config.getConfig('emoteev.overrides')));
-
-    return {
-      method: 'POST',
-      url: endpointUrl(utils.getParameterByName('emoteevEnv'), config.getConfig('emoteev.env')),
-      data: JSON.stringify(payload),
-    };
-  },
-
-  interpretResponse: (serverResponse) => serverResponse.body,
-
-  getUserSyncs: (syncOptions, serverResponses) => {
-    const parameteremoteevEnv = utils.getParameterByName('emoteev.env');
-    const configemoteevEnv = config.getConfig('emoteev.env');
-    const syncs = [];
-    if (syncOptions.iframeEnabled) {
-      syncs.push({
-        type: 'iframe',
-        url: userSyncIframeUrl(parameteremoteevEnv, configemoteevEnv),
-      });
-    }
-    if (syncOptions.pixelEnabled && serverResponses.length > 0) {
-      syncs.push({
-        type: 'image',
-        url: userSyncImageUrl(parameteremoteevEnv, configemoteevEnv),
-      });
+  isBidRequestValid: (bidRequest) => {
+    const isValid = isBidRequestValid(bidRequest);
+    if (isValid) {
+      triggerPixel(url.format(onAdapterCalled(
+        resolveEnv(config.getConfig(), getParameterByName('emoteevEnv')),
+        bidRequest)))
     }
-    return syncs;
+    return isValid;
   },
+  buildRequests: (validBidRequests, bidderRequest) =>
+    buildRequests(
+      resolveEnv(config.getConfig(), getParameterByName('emoteevEnv')),
+      resolveDebug(config.getConfig(), getParameterByName('debug')),
+      config.getConfig('currency'),
+      validBidRequests,
+      bidderRequest),
+  interpretResponse: interpretResponse,
+  onBidWon: (bidObject) =>
+    triggerPixel(url.format(onBidWon(
+      resolveEnv(config.getConfig(), getParameterByName('emoteevEnv')),
+      getCookie('_pubcid'),
+      bidObject))),
+  onTimeout: (bidRequest) =>
+    triggerPixel(url.format(onTimeout(
```

Comment:
Adapters are prohibited from firing their own pixels.  If you want fire pixels , you can return them from get getUserSyncs

Reply:
this pixel is ok since it's tied to the event handler.